### PR TITLE
Feat/stacks aggregation increase

### PR DESCRIPTION
**Pull request by jcnelson**
This PR addresses #3360.  It adds two new functions to PoX:

* `stack-aggregation-increase`.  This function _increases_ the amount of STX locked behind a PoX address that had been delegated to the caller.  This can be used to stack more pool participants' STX even if the sum of their new contributions is less than the stacking minimum.  The only requirement is that the stacking minimum be met by the PoX address when the new contributions are added.

* `stack-aggregation-commit-indexed`.  This function works exactly the same as `stack-aggregation-commit`, but it returns the PoX address index that needs to be passed to `stack-aggregation-increase` to bump the STX lock-up.

This fulfills an acceptance criterion from the technical CAB for SIP-015 posed by @friedger.  This fix makes it so stacking pool operators can increase their users' locked-up STX even if the new users' contributions don't meet the stacking minimum (this is currently required by `stack-aggregation-commit`).

### CODE REVIEW

**Review by friedger — COMMENTED** [review]
Great!
I just have two comments on the comments in the tests.

File: src/chainstate/stacks/boot/pox-2.clar [hunk 1]
```diff
@@ -419,6 +423,15 @@
     ;; It _should_ be equal to num-cycles.
     (asserts! (is-eq num-cycles (get i results)) (err ERR_STACKING_UNREACHABLE))
     (asserts! (is-eq num-cycles (len reward-set-indexes)) (err ERR_STACKING_UNREACHABLE))
+    (print {
```
Comment (jcnelson) [inline src/chainstate/stacks/boot/pox-2.clar:426]: Oops, need to drop this.

File: src/chainstate/stacks/boot/pox-2.clar [hunk 2]
```diff
@@ -442,6 +455,13 @@
           (map-set partial-stacked-by-cycle
                    { sender: tx-sender, pox-addr: pox-addr, reward-cycle: reward-cycle }
                    { stacked-amount: (+ amount-ustx current-amount) }))
+      (print {
```
Comment (jcnelson) [inline src/chainstate/stacks/boot/pox-2.clar:458]: Oops, need to drop this.

File: src/chainstate/stacks/boot/pox_2_tests.rs [hunk 3]
```diff
@@ -3838,3 +3849,452 @@ fn test_pox_2_delegate_stx_addr_validation() {
...
+
+    // our "tenure counter" is now at 10
+    assert_eq!(tip.block_height, 10 + EMPTY_SORTITIONS as u64);
+
+    let mut latest_block = peer.tenure_with_txs(
+        &[alice_delegation_1, delegate_stack_tx_bob, stack_tx_dan],
+        &mut coinbase_nonce,
+    );
+
+    // check that the partial stacking state contains entries for bob and charlie
```
[the remark this answers is not in the document]
Comment (jcnelson) [reply src/chainstate/stacks/boot/pox_2_tests.rs:3985]: Yup, that's an artifact from an earlier version of this test that was actually unworkable

File: src/chainstate/stacks/boot/pox_2_tests.rs [hunk 4]
```diff
@@ -3838,3 +3849,452 @@ fn test_pox_2_delegate_stx_addr_validation() {
...
+        latest_block = peer.tenure_with_txs(&[], &mut coinbase_nonce);
+    }
+
+    let alice_bal = get_stx_account_at(&mut peer, &latest_block, &alice_principal);
+    assert_eq!(alice_bal.amount_locked(), alice_first_lock_amount);
+
+    let dan_bal = get_stx_account_at(&mut peer, &latest_block, &dan_principal);
+    assert_eq!(dan_bal.amount_locked(), dan_stack_amount);
+
+    // check that the partial stacking state contains entries for bob
```
[the remark this answers is not in the document]
Comment (jcnelson) [reply src/chainstate/stacks/boot/pox_2_tests.rs:4011]: Yup, will update

**Review by obycode — APPROVED** [review]
LGTM

**Comment by jcnelson** [comment]
Yup, reproduced locally

**Comment by jcnelson** [comment]
Fixed -- problem was that the chains-coordinator thread didn't have the same stack size as the miner (2MB vs 32MB), so when we stress-test processing an abnormally-large transaction (which wouldn't even be allowed under the consensus rules), it overflowed.

**Pull request merged**